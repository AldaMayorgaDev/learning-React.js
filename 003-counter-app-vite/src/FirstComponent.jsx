import PropTypes from 'prop-types';

const getSaludo = ()=>{
  return 'Hola Aldahir';
}
export const FirstComponent = ({title, mensaje= 'Legend of zelda Tears of the Kingdom',edad  }) => {
  
  if (!title) {
    throw new Error('el title no existe.')
  }
  
  console.log('props.title', title)
  return (
    <>
      <h2>{title}</h2>
      <h3>{mensaje}</h3>
      <p>{getSaludo()}</p>
      <p>{edad + 10}</p>
    </>
  )
};

FirstComponent.propTypes = {
    title: PropTypes.string.isRequired,
    mensaje: PropTypes.string,
    edad : PropTypes.number.isRequired
}

FirstComponent.defaultProps = {
  title: 'No hay titulo',
  edad: 0,
  mensaje: 'The Legend of Zelda: Tears of the kingdom'
}