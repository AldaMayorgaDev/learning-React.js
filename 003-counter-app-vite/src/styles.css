html, body {
    background-color: #21232A;
    color: white;
    font-family: Helvetica, arial, sans-serif;
    font-size: 16px;
    padding: 70px;
}

button {
    padding:15px;
    font-size: 1.3rem;
    border-radius: .5rem;
    margin-right: 15px;
}