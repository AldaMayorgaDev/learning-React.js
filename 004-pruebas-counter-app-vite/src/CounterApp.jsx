//importando hooks
import { useState } from 'react';

//Importando Fragment solo pro practica puede remplazarse por el uso de <></>
// Importando PropTypes para definir tipo de datos de las propiedades
import { Fragment } from "react";
import PropTypes from "prop-types";


//functional componente CounterApp que recibe value

export const CounterApp =({value})=>{

    const [ counter , setCounter ] = useState ( value );


    const handleAdd = ()=>{ 
    
        setCounter( counter +1 );
    }

    const handleSubtract = ()=>{
        setCounter( counter - 1);
    }

    const handleReset = ()=>{
        setCounter( value );
    }
    return (
        <Fragment>
            <h1>CouterApp</h1>
            <h2>{ counter }</h2>
            <button onClick={  ( )=>{ handleAdd()}  }>
                +1
            </button>

            <button onClick={handleSubtract}>
                 -1 
            </button>


            <button aria-label="btn-reset" onClick={handleReset}> 
                Reset 
            </button>
        </Fragment>
    );
}

// Definicion de propTypes para value
CounterApp.propTypes ={
    value: PropTypes.number.isRequired
}

// Definicion de default props
CounterApp.defaultProps = {
    value: 0,
}