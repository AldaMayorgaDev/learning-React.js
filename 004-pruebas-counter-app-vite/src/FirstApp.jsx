
//Exportando componente
export function App() {
    return <h1>Hola Mundo:!</h1>;
}


//Exportando componente Ocupando el shorthoot rafc
export const FirstApp = () => {
  return (
    <h1>Hello word</h1>
  )
}
