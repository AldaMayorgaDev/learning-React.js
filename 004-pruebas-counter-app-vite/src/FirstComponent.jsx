import PropTypes from 'prop-types';


export const FirstComponent = ({title, subtitle,name  }) => {
  
  return (
    <>
      <h1 data-testid="test-title">{title}</h1> {/* Se a√±ade el atibuto data-testid para buscar en las pruebas con getByTestId() */}
      <p>{subtitle}</p>
      <p>{subtitle}</p>
      <p>{name }</p>
    </>
  )
};

FirstComponent.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string,
    name : PropTypes.string.isRequired
}

FirstComponent.defaultProps = {
  name: 'Aldahir',
  subtitle: 'No hay subtitulo',
}