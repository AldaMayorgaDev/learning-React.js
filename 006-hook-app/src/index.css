body{
    padding: 20px;
}

button{
    margin-right: 10px;
}

/* a{
    color: #131313;
    text-decoration: none;
}

a:active{
    color: #69696A;
    text-decoration: none;
} */

/* .navbar-brand, .nav-link{
    padding-left: 10px
} */