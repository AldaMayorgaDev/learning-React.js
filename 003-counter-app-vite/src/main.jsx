import React from 'react';
import ReactDOM from 'react-dom/client';
//import {App, FirstApp} from './FirstApp';
//import { FirstComponent } from './FirstComponent';
import { CounterApp } from './CounterApp';

//Importar estilos
import './styles.css';


ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp  value={1}/>
    </React.StrictMode>
);