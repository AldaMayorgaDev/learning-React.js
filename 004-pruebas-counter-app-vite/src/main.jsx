import React from 'react';
import ReactDOM from 'react-dom/client';
//import {App, FirstApp} from './FirstApp';
//import { FirstComponent } from './FirstComponent';
import { CounterApp } from './CounterApp';

//Importar estilos
import './styles.css';


ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp  value={100}/>
       {/*  <FirstComponent title="Legend of zelda: tears of the kingdom" /> */}
    </React.StrictMode>
);