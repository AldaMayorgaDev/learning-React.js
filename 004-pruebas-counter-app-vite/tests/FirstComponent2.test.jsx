import { render, screen } from '@testing-library/react';
import { FirstComponent } from '../src/FirstComponent';

describe('Pruebas en <FirstComponent />', () => { 
    const title = 'Lengend of Zelda';
    const subtitle = 'Tears of the kingdom';

    test('Debe hacer Match con el Snapshot', ()=>{

        const {container} = render(<FirstComponent title={title}/>);
        expect(container).toMatchSnapshot();
    });


    test('Debe de mostrar el mensaje: "Legend od Zelda"', () => { 

        render(<FirstComponent title={title}/>);
        expect( screen.getByText(title) ).toBeTruthy();
    });

    test('Debe de mostrar el titlo en un h1', ()=>{ 
        render(<FirstComponent title={title}/>);
        expect( screen.getByRole('heading', {level: 1}).innerHTML).toContain(title);
    });


    test('Debe mostrar el subtitulo enviado por props', () => { 
        render(<FirstComponent title={title} subtitle={subtitle}/>);

        expect(screen.getAllByText(subtitle).length).toBe(2);
    });
 })