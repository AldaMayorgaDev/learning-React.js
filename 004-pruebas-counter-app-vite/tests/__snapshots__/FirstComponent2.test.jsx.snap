// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstComponent /> Debe hacer Match con el Snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Lengend of Zelda
  </h1>
   
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
  <p>
    Aldahir
  </p>
</div>
`;
